src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerReleaseService.java

```diff
@@ -0,0 +1,137 @@
+import uk.gov.justice.hmpps.prison.repository.jpa.model.MovementTypeAndReason.Pk;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.AgencyInternalLocationRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.AgencyLocationRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.BedAssignmentHistoriesRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.ExternalMovementRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.MovementTypeAndReasonRespository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderBookingRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderKeyDateAdjustmentRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderSentenceAdjustmentRepository;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.ReferenceCodeRepository;
+import uk.gov.justice.hmpps.prison.security.AuthenticationFacade;
+
+import java.time.LocalDateTime;
+
+import static uk.gov.justice.hmpps.prison.repository.jpa.model.MovementType.REL;
+
+@Service
+@Transactional
+@Validated
+@Slf4j
+@AllArgsConstructor
+public class PrisonerReleaseService {
+
+    private final OffenderBookingRepository offenderBookingRepository;
+    private final AgencyLocationRepository agencyLocationRepository;
+    private final ExternalMovementRepository externalMovementRepository;
+    private final AgencyService agencyService;
+
+    private final ReferenceCodeRepository<MovementType> movementTypeRepository;
+    private final ReferenceCodeRepository<MovementReason> movementReasonRepository;
+    private final BedAssignmentHistoriesRepository bedAssignmentHistoriesRepository;
+    private final AgencyInternalLocationRepository agencyInternalLocationRepository;
+    private final MovementTypeAndReasonRespository movementTypeAndReasonRespository;
+    private final OffenderSentenceAdjustmentRepository offenderSentenceAdjustmentRepository;
+    private final OffenderKeyDateAdjustmentRepository offenderKeyDateAdjustmentRepository;
+
+    public void releasePrisoner(final String prisonerIdentifier, final String movementReasonCode, final String commentText) {
+
+        // check that prisoner is active in
+        final var optionalOffenderBooking = offenderBookingRepository.findByOffenderNomsIdAndBookingSequence(prisonerIdentifier, 1);
+
+        final var booking = optionalOffenderBooking.orElseThrow(EntityNotFoundException.withMessage("No bookings found for prisoner number %s", prisonerIdentifier));
+
+        if (!booking.getActiveFlag().equals("Y")) {
+            throw new BadRequestException("Prisoner is not currently active");
+        }
+
+        if (!booking.getInOutStatus().equals("IN")) {
+            throw new BadRequestException("Prisoner is not currently IN");
+        }
+
+        // unless you are system user you must have the caseload
+        if (!AuthenticationFacade.hasRoles("SYSTEM_USER")) {
```

Comment:
Can this be done with @VerifyBookingAccess (which accepts an override role I think)?

Reply:
Already done this bit in the resource class - removed